docs/contracts.rst

```diff
@@ -733,6 +732,61 @@ For example:
        >>> rich_logs[0]['args']
        {'myArg': 12345}
 
+   By default, an error will be raised if the log doesn't match the provided ABI.
+   However, errors can be optionally ignored by passing in the ``ignore`` or
+   ``discard`` flag located in the ``EventLogErrorFlags`` class. The ``discard``
+   flag will discard any logs that raise an error during processing, and return
+   the rest. The ``ignore`` flag will return the raw log that caused the error,
+   without any processing, and will process any logs that do not raise an error.
+   An ``EventLogErrorFlag`` is imported from web3/_utils/events.py.
```

Comment:
I'm realizing that we *probably* need to think about the real use cases with these.  Here are my thoughts.

- `discard` might be considered scary but probably useful one.
- `warn` might be a good new one that raises a warning and discards the log
- `ignore` probably needs a mechanism for reliable programatic differentiation which logs are properly processed and which logs threw errors.

Reply:
Yeah, agreed. I like the `warn` option.

For `ignore`, maybe we would return the log, with an additional errors field that we add? So the returned log might look like: 
```
["logs": {
    "address": "0x00000000000000000000",
    ...,
    "type": "mined",
    "errors": "ValueError: <error message>"
}]
```

Reply:
seems reasonable.